msal4j-sdk/src/samples/msi-sample/src/main/java/com/microsoft/identity/msi/HttpTriggerJava.java

```diff
@@ -0,0 +1,52 @@
+package com.microsoft.identity.msi;
+
+import java.util.*;
+
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+import com.microsoft.aad.msal4j.ManagedIdentityApplication;
+import com.microsoft.aad.msal4j.ManagedIdentityId;
+import com.microsoft.aad.msal4j.ManagedIdentityParameters;
+import com.microsoft.azure.functions.annotation.*;
+import com.microsoft.azure.functions.*;
+
+/**
+ * Azure Function with HTTP Trigger to test managed identity.
+ */
+public class HttpTriggerJava {
+    /**
+     * This function listens at endpoint "/api/MsiFunctionJava". Two ways to invoke it using "curl" command in bash:
+     * 1. curl -d "HTTP Body" {your host}/api/MsiFunctionJava
+     * 2. curl {your host}/api/MsiFunctionJava?resource=HTTP%20Query
+     */
+    @FunctionName("MsiFunctionJava")
+    public HttpResponseMessage run(
+            @HttpTrigger(name = "req", methods = {HttpMethod.GET, HttpMethod.POST}, authLevel = AuthorizationLevel.FUNCTION) HttpRequestMessage<Optional<String>> request,
+            final ExecutionContext context) {
+        context.getLogger().info("Java HTTP trigger processed a request.");
+
+        // Parse query parameter
+        String query = request.getQueryParameters().get("resource");
+        String resource = request.getBody().orElse(query);
+
+        if (resource == null) {
+            return request.createResponseBuilder(HttpStatus.BAD_REQUEST).body("Please pass a resource to acquire token for").build();
+        }
+
+        String response;
+
+        try {
+            ManagedIdentityApplication msiApp = ManagedIdentityApplication
+                    .builder(ManagedIdentityId.systemAssigned())
+                    .logPii(true)
```

Comment:
You can find some docs on logging here (it includes a step about spring for some reason, I think that's a mistake): https://learn.microsoft.com/en-us/entra/msal/java/advanced/msal-logging-java

In short, whenever we want to to log something we do it through the [Simple Logging Facade for Java](http://www.slf4j.org/). You've probably seen code like `LOG.info("abc")` and `LOG.error("xyz")`, those are all getting sent through SLF4J. An app developer can use their preferred logging framework, and as long as it's compatible with SLF4J then it will output our logs through that framework in the same way as their own logs.